cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -162,12 +159,113 @@ def __init__(self, scope: core.Construct, construct_id: str, imagebuild: ImageBu
-            description="Image Id for PCluster",
-            name="-".join(["/Test/Images/PCluster", resources_prefix]),
-            type="String",
-            value=core.Fn.get_att("PClusterImage", "ImageId").to_string(),
+            id="InstanceRole",
+            managed_policy_arns=managed_policy_arns,
+            assume_role_policy_document={
+                "Statement": {
+                    "Action": "sts:AssumeRole",
+                    "Effect": "Allow",
+                    "Principal": {"Service": "ec2.amazonaws.com"},
+                },
+                "Version": "2012-10-17",
+            },
+            path="/executionServiceEC2Role/",
         )
+
+        instancerole.add_metadata("Comment", "Role to be used by instance during image build.")
+
+    def _set_instance_profile(self, instance_role="InstanceRole"):
+        """Set default instance profile in imagebuilder cfn template."""
+        iam.CfnInstanceProfile(
+            self, id="InstanceProfile", path="/executionServiceEC2Role/", roles=[core.Fn.ref(instance_role)]
+        )
+
+    def _set_custom_components(self, components):
+        """Set custom component in imagebuilder cfn template."""
+        build = self.imagebuild.build
+        custom_components = build.components
+        initial_component_len = len(custom_components)
+        arn_components_len = 0
+        for custom_component in custom_components:
+            if custom_component.type.startswith("arn"):
+                components.append(
+                    imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(component_arn=custom_component)
+                )
+                arn_components_len += 1
+            elif custom_component.type.startswith("yaml"):
+                imagebuilder.CfnComponent(
+                    self,
+                    id="CustomComponent" + str(len(components) - initial_component_len - arn_components_len),
+                    name="-".join(
+                        ["CustomComponent", str(len(components) - initial_component_len - arn_components_len)]
+                    ),
+                    version="0.0.1",
+                    description="CustomComponent",
+                    change_description="First version",
+                    platform="Linux",
+                    uri=custom_component,
+                )
+            else:
+                imagebuilder.CfnComponent(
+                    self,
+                    id="CustomComponent" + str(len(components) - initial_component_len - arn_components_len),
+                    name="-".join(
+                        ["CustomComponent", str(len(components) - initial_component_len - arn_components_len)]
+                    ),
+                    version="0.0.1",
+                    description="CustomComponent",
+                    change_description="First version",
+                    platform="Linux",
+                    data=self._wrap_bash_to_component(),
```

Comment:
`_wrap_bash_to_component` is not implemented yet? Let's add a #TODO label

Reply:
Add #TODO